compiler/qsc_codegen/src/qir.rs

```diff
@@ -7,11 +7,59 @@ mod instruction_tests;
 #[cfg(test)]
 mod tests;
 
+use qsc_frontend::compile::RuntimeCapabilityFlags;
+use qsc_hir::hir;
+use qsc_lowerer::map_hir_package_to_fir;
+use qsc_partial_eval::partially_evaluate;
 use qsc_rir::{
+    passes::{check_and_transform, defer_quantum_measurements},
     rir::{self, ConditionCode},
     utils::get_all_block_successors,
 };
 
+fn lower_store(package_store: &qsc_frontend::compile::PackageStore) -> qsc_fir::fir::PackageStore {
```

Comment:
There are a couple of `lower_store` methods in multiple crates that duplicate this functionality. It would be worth it to implement it in the new `qsc_lowerer` crate and use it from there.

Reply:
It can't be implemented in the lowerer crate due to extra deps. This is bigger refactoring that we can look at later.